OloEngine/src/OloEngine/Renderer/MeshOptimization.cpp

```
@@ -0,0 +1,528 @@
+        {
+            return;
+        }
+
+        auto vertexCount = static_cast<sizet>(vertices.Num());
+        auto indexCount = static_cast<sizet>(indices.Num());
+
+        // 1. Vertex cache optimization — reorder triangles for GPU post-transform cache
+        meshopt_optimizeVertexCache(indices.GetData(), indices.GetData(), indexCount, vertexCount);
+
+        // 2. Overdraw optimization — reorder triangles to reduce pixel overdraw
+        constexpr f32 kOverdrawThreshold = 1.05f;
```

Clang (14.0.6)

[warning] variable 'kOverdrawThreshold' is not initialized

(cppcoreguidelines-init-variables)